packages/shorebird_cli/lib/src/commands/patch_new/android_patcher.dart

```diff
@@ -0,0 +1,177 @@
+        );
+      exit(ExitCode.software.code);
+    }
+    return aabFile;
+  }
+
+  @override
+  Future<Map<Arch, PatchArtifactBundle>> createPatchArtifacts({
+    required String appId,
+    required int releaseId,
+  }) async {
+    final releaseArtifacts = await codePushClientWrapper.getReleaseArtifacts(
+      appId: appId,
+      releaseId: releaseId,
+      architectures: AndroidArch.availableAndroidArchs,
+      platform: releaseType.releasePlatform,
+    );
+    final releaseArtifactPaths = <Arch, String>{};
+    final downloadReleaseArtifactProgress = logger.progress(
+      'Downloading release artifacts',
+    );
+
+    for (final releaseArtifact in releaseArtifacts.entries) {
+      try {
+        final releaseArtifactFile = await artifactManager.downloadFile(
+          Uri.parse(releaseArtifact.value.url),
+        );
+        releaseArtifactPaths[releaseArtifact.key] = releaseArtifactFile.path;
+      } catch (error) {
+        downloadReleaseArtifactProgress.fail('$error');
+        exit(ExitCode.software.code);
+      }
+    }
+
+    downloadReleaseArtifactProgress.complete();
+
+    final patchArchsBuildDir = ArtifactManager.androidArchsDirectory(
+      projectRoot: projectRoot,
+    );
+    if (patchArchsBuildDir == null) {
+      logger.err('Could not find patch artifacts');
+      exit(ExitCode.software.code);
+    }
+
+    final patchArtifactBundles = <Arch, PatchArtifactBundle>{};
+    final createDiffProgress = logger.progress('Creating patch artifacts');
+    for (final releaseArtifactPath in releaseArtifactPaths.entries) {
+      final arch = releaseArtifactPath.key;
+      final patchArtifactPath = p.join(
+        patchArchsBuildDir.path,
+        arch.androidBuildPath,
+        'libapp.so',
+      );
+      logger.detail('Creating artifact for $patchArtifactPath');
+      final patchArtifact = File(patchArtifactPath);
+      final hash = sha256.convert(await patchArtifact.readAsBytes()).toString();
+      try {
+        final diffPath = await artifactManager.createDiff(
+          releaseArtifactPath: releaseArtifactPath.value,
+          patchArtifactPath: patchArtifactPath,
+        );
+        patchArtifactBundles[releaseArtifactPath.key] = PatchArtifactBundle(
+          arch: arch.arch,
+          path: diffPath,
+          hash: hash,
+          size: await File(diffPath).length(),
+        );
+      } catch (error) {
+        createDiffProgress.fail('$error');
+        exit(ExitCode.software.code);
+      }
+    }
+    createDiffProgress.complete();
+    return patchArtifactBundles;
+  }
+
+  @override
+  Future<String> extractReleaseVersionFromArtifact(File artifact) async {
+    return shorebirdAndroidArtifacts.extractReleaseVersionFromAppBundle(
+      artifact.path,
+    );
+  }
+
+  @override
+  Future<File> patchArtifactForDiffCheck() async {
+    return shorebirdAndroidArtifacts.findAab(
+      project: projectRoot,
+      flavor: flavor,
+    );
```

Comment:
FYI, if we don't await the future here, this call won't show up in the call stack, only the call where it is actually awaited will (that may not matter).

Reply:
`findAab` doesn't return a future, so nothing to await

Reply:
Also turns out to not be needed. Deleted! :)